### DISCUSSION TITLE
Naming conventions for schema and data files

### DISCUSSION THREAD

**Discussion by olliefr** [valocode/bubbly#47] [Ideas]
Currently, we have a number of ways we name schemas and mock data files in our tests. I propose that we agree on a single style. Because consistency rules, obviously.

I think this is a good question for the Discussion format?

## Current situation

For the schema we have:

* `schema.bubbly` in `integration/testdata/schema/`
* `tables.hcl` in `store/testdata/`
* `tables[0-5].hcl` in `store/testdata/sqlgen/`
* `tutorial1.schema` in `website/docs/tutorials/`

For the data we have:

* `data.hcl` in `store/testdata/`
* `data[0-5].hcl` in `store/testdata/sqlgen/`

### 1. File name extension

One aspect of this is the file name extension: do we want to use custom `.bubbly` or generic `.hcl`?

➕ of `.bubbly` is visibility.
➖ of `.bubbly` is no syntax highlighting by default.

➕ of `.hcl` is the syntax highlighting turned on by default in VSCode and GitHub.

Examples:

* No syntax highlighting in GitHub: [`schema.bubbly`][nosyn]
* Syntax highlighting is **on** in GitHub: [`tables.hcl`][yesyn]

_My_ vote is for the generic extension `.hcl` as this would guarantee syntax highlighting in various editors and other tools, both online and offline.

### 2. File name

For schema, we have two obvious choices: `schema` and `tables`. My vote is for `schema`, as the _Bubbly Schema_ is one of the core concepts in Bubbly.

For data, an obvious choice is `data` and we have little diversity on this now, so it's not really an issue.

## Proposed action

I am, obviously, open to other ideas!

But, if there is no strong objections, I propose to:

- [ ] Rename `schema.bubbly` to `schema.hcl` in `integration/testdata/schema/`
- [ ] Rename `tables.hcl` to `schema.hcl` in `store/testdata/`
- [ ] Rename `tables[0-5].hcl` to `schema[0-5].hcl` in `store/testdata/sqlgen/`
- [ ] Rename `tutorial1.schema` to `tutorial-1-schema.hcl` in `website/docs/tutorials/`

And to stick to `schema{.*}.hcl` for our other schema adventures 🙃 

What do you think, @benmarsden @jlarfors @nate-droid ?

[nosyn]: https://github.com/valocode/bubbly/blob/35f61e7150e5019335232ae2d72f1be57856df46/integration/testdata/schema/schema.bubbly
[yesyn]: https://github.com/valocode/bubbly/blob/35f61e7150e5019335232ae2d72f1be57856df46/store/testdata/tables.hcl

**Comment by nate-droid** [comment]
At least in Goland, you are able to associate .bubbly files with HCL, which brings along all of the nice default syntax highlighting and code completion. But I like the idea of having a consistent naming convention for sure! +1 for schema instead of tables :)

**Reply by olliefr** [reply]
You can do this in VSCode as well, but that's something that does not happen automatically.

And I'm not sure, if GitHub can be convinced at all to render `.bubbly` files using HCL syntax highlighting.
